sabnzbd/nzbqueue.py

```diff
@@ -813,12 +813,13 @@ def register_article(self, article, found=True):
         else:
             if file_done:
                 if nzo.next_save is None or time.time() > nzo.next_save:
-                    sabnzbd.save_data(nzo, nzo.nzo_id, nzo.workpath)
                     BPSMeter.do.save()
                     if nzo.save_timeout is None:
                         nzo.next_save = None
                     else:
                         nzo.next_save = time.time() + nzo.save_timeout
+                    # After setting next_save, to avoid threading issues!
+                    sabnzbd.save_data(nzo, nzo.nzo_id, nzo.workpath)
```

Comment:
Hmmm, yes you are right. I was mixed things up. I read python does the file-wryting in a seperate process, but it still locks the current thread for that. I will revert this.